geopandas/io/file.py

```diff
@@ -57,6 +57,8 @@ def read_file(filename, bbox=None, mask=None, rows=None, **kwargs):
     rows: int or slice, default None
         Load in specific rows by passing an integer (first `n` rows) or a
         slice() object.
+    columns : list, optional
```

Comment:
@snowman2 suggestion: there may be use cases of reading a subset of columns and not the geometry column - though the return in that case should probably not be a GeoDataFrame.  For example, reading tabular data that requires a geo driver, such as a table in an ESRI File Geodatabase, but where the table either doesn't include geometry or the user doesn't want it.

As such, it might be appropriate to include `'geometry'` as a named field accessible in columns, and if columns is not `None` but omits `geometry`, raise a warning and return a DataFrame of the specified columns?

From a data frame perspective, `geometry` is a column too, so it feels a little odd to not consider it a column from an I/O perspective.

Also note that Fiona has a concept of ignored fields [here](https://github.com/Toblerity/Fiona/blob/1.8.13/fiona/collection.py#L37).  You'd have to inspect the schema of the data source from Fiona to convert your white-listed list of columns into a black-listed list of columns to ignore.  This would allow you to avoid reading all the fields from the source then ignoring some in Python; the ignored fields in Fiona get applied in the Cython tier directly against the OGR data source and should be faster.  

Likewise, you could leverage `ignore_geometry` here too if `geometry` is absent from `columns`.


Aside:
Might be good to note the default case here too, `Default behavior is to load all columns from the file.`

Reply:
Just updated and added support for `ignore_geometry`.